src/main/java/com/actiontech/dble/backend/mysql/nio/handler/transaction/refactor/stage/XARollbackStage.java

```diff
@@ -0,0 +1,90 @@
+package com.actiontech.dble.backend.mysql.nio.handler.transaction.refactor.stage;
+
+import com.actiontech.dble.backend.mysql.nio.MySQLConnection;
+import com.actiontech.dble.backend.mysql.nio.handler.transaction.refactor.XATransactionContext;
+import com.actiontech.dble.backend.mysql.xa.TxState;
+import com.actiontech.dble.backend.mysql.xa.XAStateLog;
+import com.actiontech.dble.btrace.provider.XaDelayProvider;
+import com.actiontech.dble.route.RouteResultsetNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class XARollbackStage extends AbstractXAStage {
+
+    private static Logger logger = LoggerFactory.getLogger(XARollbackStage.class);
+    private boolean lastStageIsXAEnd;
+
+
+    public XARollbackStage(XATransactionContext context, boolean isFromEndStage) {
+        super(context);
+        this.lastStageIsXAEnd = isFromEndStage;
+    }
+
+    @Override
+    public XAStage next(boolean isFail) {
+        if (isFail) {
```

Comment:
lastStageIsXAEnd must be used in this return function, otherwise logically incomplete